fem/fespace.cpp

```diff
@@ -1632,167 +1879,406 @@ void FiniteElementSpace::Construct()
    // This method should be used only for non-NURBS spaces.
    MFEM_VERIFY(!NURBSext, "internal error");
 
+   // Variable order space needs a nontrivial P matrix + also ghost elements
+   // in parallel, we thus require the mesh to be NC.
+   MFEM_VERIFY(!IsVariableOrder() || Nonconforming(),
+               "Variable order space requires a nonconforming mesh.");
+
    elem_dof = NULL;
-   bdrElem_dof = NULL;
+   bdr_elem_dof = NULL;
    face_dof = NULL;
 
    ndofs = 0;
-   nedofs = nfdofs = nbdofs = 0;
+   nvdofs = nedofs = nfdofs = nbdofs = 0;
    bdofs = NULL;
-   fdofs = NULL;
+
    cP = NULL;
    cR = NULL;
+   cR_hp = NULL;
    cP_is_set = false;
    // 'Th' is initialized/destroyed before this method is called.
 
-   nvdofs = mesh->GetNV() * fec->DofForGeometry(Geometry::POINT);
+   int dim = mesh->Dimension();
+   int order = fec->GetOrder();
+
+   MFEM_ASSERT(mesh->GetNumGeometries(dim),
+               "Mesh was not correctly finalized.");
```

Comment:
I think it can be `MFEM_VERIFY` it can give valuable information without impacting the performance.

Reply:
Agreed.

Reply:
This was a good suggestion that revealed several cases where an unfinalized Mesh was being used... :+1: